util/dvsim/CircularList.py

```diff
@@ -0,0 +1,142 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+
+class Node:
+    '''Implements a node in a linked list.'''
+    def __init__(self, item=None, next_node=None):
+        self.item = item
+        self.next_node = next_node
+
+    def __str__(self):
+        return self.item.__str__()
+
+    def __repr__(self):
+        return self.item.__repr__()
```

Comment:
Is this needed? If not, could you drop it? Hacking `repr` with something that doesn't read as the object you want causes all sorts of problems.

Reply:
Done.